src/models/person/__init__.py

```diff
@@ -2,29 +2,43 @@
 
 from pydantic import BaseModel
 
+from src.models.wikimedia.wikipedia.templates.enums import WikipediaTemplatePersonRole
+
 
 class Person(BaseModel):
     """Model a person mentioned in a page_reference
     Sometimes they are stated as "editor=John Niel"
     and we save that as name_string for later disambiguation"""
 
     given: Optional[str]
-    has_number: bool
     link: Optional[str]
     mask: Optional[str]
     name_string: Optional[str]
     number_in_sequence: Optional[int]
     orcid: Optional[str]
-    role: Any
+    # Pydantic 1.9.0 cannot handle hiearchical enums it seems so we use Any here
+    role: Any  # type: WikipediaTemplatePersonRole
     surname: Optional[str]
 
     @property
-    def author_name_string(self) -> Optional[str]:
+    def full_name(self) -> str:
         """We hardcode western cultural name ordering pattern here with the
         order "givenname surname".  We use str.strip() because no name has
-        significant whitespace at the beginning or end of the string"""
-        return (
-            (self.name_string or "").strip()
-            or f"{self.given or ''} {self.surname or ''}".strip()
-            or None
-        )
+        significant whitespace at the beginning or end of the string
+
+        We want this property to return either the full name or empty string"""
+        if self.name_string:
```

Comment:
What if self.name_string == " " or "   "

Reply:
Pretty sure that is impossible. The template parsing ignores empty parameters and space is a separator = empty. We should test that BTW.  E.g. {{cite journal | given=  | surname=test}} -> Person(given=None,surname="test") -> full_name()-> ???

Reply:
A tab character "\t" or a return character "\n" are also whitespace that .strip() should eliminate.